tools/odbc/statement_functions.cpp

```diff
@@ -216,7 +238,12 @@ SQLRETURN duckdb::GetDataStmtResult(SQLHSTMT statement_handle, SQLUSMALLINT col_
 			return SQL_SUCCESS;
 		}
 		case SQL_C_CHAR: {
-			auto out_len = snprintf((char *)target_value_ptr, buffer_length, "%s", val.GetValue<string>().c_str());
+			std::string val_str = val.GetValue<std::string>();
+			if (!target_value_ptr) {
+				return SetStringValueLength(val_str, str_len_or_ind_ptr);
+			}
+
+			auto out_len = snprintf((char *)target_value_ptr, buffer_length, "%s", val_str.c_str());
```

Comment:
`snprintf` seems overkill here no? Can't we just set `out_len` to `MinValue(val_str.size(), buffer_length)` and use `memcpy`?

Reply:
Yeah, that makes sense.